libshaderc/src/shaderc_test.cc

```diff
@@ -232,6 +232,42 @@ TEST(CompileStringWithOptions, IfDefCompileOption) {
                                  shaderc_glsl_vertex_shader, options.get()));
 }
 
+TEST(CompileStringWithOptions, MessageRules) {
+  Compiler compiler;
+  compile_options_ptr options(shaderc_compile_options_initialize());
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+
+  // Confirm that this shader only compiles with shaderc_msg_default;
+  // if using SPIR-V rules, glslang will fail to compile it
+  const std::string kGlslShader =
+    R"(#version 100
+       uniform highp sampler2D tex;
+       void main() {
+         gl_FragColor = texture2D(tex, vec2(0.0,0.0));
+       }
+  )";
+
+  EXPECT_FALSE(CompilesToValidSpv(compiler.get_compiler_handle(),
+                                  kGlslShader,
+                                  shaderc_glsl_fragment_shader, options.get()));
+
+  shaderc_compile_options_set_message_rules(options.get(), shaderc_msg_default);
+  EXPECT_TRUE(CompilesToValidSpv(compiler.get_compiler_handle(),
+                                 kGlslShader,
+                                 shaderc_glsl_fragment_shader, options.get()));
+
+  shaderc_compile_options_set_message_rules(options.get(), shaderc_msg_spirv_rules);
+  EXPECT_FALSE(CompilesToValidSpv(compiler.get_compiler_handle(),
+                                  kGlslShader,
+                                  shaderc_glsl_fragment_shader, options.get()));
+
+  shaderc_compile_options_set_message_rules(options.get(),
+                                            static_cast<shaderc_msg_rules>(shaderc_msg_spirv_rules | shaderc_msg_vulkan_rules));
```

Comment:
As above, this again stems from glslang and EShMessages. While I could solve the immediate problem by adding a third member to the enum, my concern is making the API future-proof. There are other glslang EShMessages options we may choose to expose in the future. Alternatively, I could see glslang adding an option down the road to generate different versions of SPIR-V for example.

I see two main alternatives here:

1) change the shaderc_msg_* values to #defines,  and pass in an unsigned int for the parameter. That's less clean, but makes it easier to expose additional glslang options in the future

2) change the enum values to entries to shaderc_msg_default, shaderc_msg_spirv_glsl_rules and shaderc_msg_spirv_vulkan_rules. This is cleaner and fulfills the immediate need, but doesn't leave an easy avenue for exposing additional glslang options without adding additional API entrypoints

Since you're looking to break from the underlying glslang/EShMessages pattern, which approach do you prefer?